include/async_scope.hpp

```diff
@@ -708,180 +364,283 @@ namespace _P2519::execution {
+                    private:
+                        friend receiver_adaptor<__receiver>;
+                        template<class... __An>
+                        void set_value(__An&&... an) noexcept {
+                            _P2300::execution::set_value(std::move(__op_->__rcvr_), an...);
+                        }
+                        template<class __Err>
+                        void set_error(__Err&& __e) noexcept {
+                            _P2300::execution::set_error(std::move(__op_->__rcvr_), __e);
+                        }
+                        void set_stopped() noexcept {
+                            _P2300::execution::set_stopped(std::move(__op_->__rcvr_));
+                        }
+                        auto get_env() const& {
+                          return make_env(with(_P2300::execution::get_stop_token, __get_stop_token_(__get__scope_(__op_))));
+                        }
+                    };
+                    [[no_unique_address]] __Receiver __rcvr_;
+                    [[no_unique_address]] connect_result_t<__Constrained, __receiver> __op_;
+                    template<class _Constrained, class _Receiver>
+                      explicit __op(async_scope* __scope, _Constrained&& __c, _Receiver&& __r) 
+                      : __op_base(__scope)
+                      , __rcvr_((_Receiver&&)__r)
+                      , __op_(connect((_Constrained&&)__c, __receiver{this})) {}
+                private:
+                    void notify_waiter() noexcept override {
+                        start(__op_);
+                    }
+
+                    friend void tag_invoke(start_t, __op& __self) noexcept {
+                        std::unique_lock __guard{__self.__scope_->__lock_};
+                        auto& __active = __self.__scope_->__active_;
+                        auto& __waiters = __self.__scope_->__waiters_;
+                        if (__active != 0) {
+                            __waiters.push_back(&__self);
+                            return;
+                        }
+                        __guard.unlock();
+                        start(__self.__op_);
+                    }
+                };
+                async_scope* __scope_;
+                [[no_unique_address]] __Constrained __c_;
+            private:
+                template <__decays_to<__sender> _Self, class _Receiver>
+                  [[nodiscard]] friend __op<__x<remove_cvref_t<_Receiver>>> tag_invoke(connect_t, _Self&& __self, _Receiver&& __rcvr) {
+                    return __op<__x<remove_cvref_t<_Receiver>>>{__self.__scope_, ((_Self&&) __self).__c_, (_Receiver&&)__rcvr};
+                  }
+                template <__decays_to<__sender> _Self, class _Env>
+                  friend auto tag_invoke(get_completion_signatures_t, _Self&&, _Env)
+                    -> completion_signatures_of_t<__member_t<_Self, __Constrained>, _Env>;
+            };
+        };
+        template<class _Constrained>
```

Comment:
What are the type requirements on `_Constrained`?

Reply:
sender